src/content/docs/analyzer/plugins.mdx

```diff
@@ -0,0 +1,52 @@
+---
+title: Analyzer Plugins
+description: An overview of how to use Biome's Analyzer Plugins
+---
+
+:::note
+Plugins are currently only available in the [2.0 beta release](/blog/biome-v2-0-beta).
+:::
+
+Biome Analyzer supports [GritQL](/reference/gritql/) plugins. Currently, these
+plugins allow you to match specific code patterns and register customized
+diagnostic messages for them.
+
+An example of a plugin is:
+
+```grit
+`$fn($args)` where {
+    $fn <: `Object.assign`,
+    register_diagnostic(
+        span = $fn,
+        message = "Prefer object spread instead of `Object.assign()`"
+    )
+}
+```
+
+You can put a GritQL snippet in a file anywhere in your project, but be mindful
+you use the `.grit` extension. Then, you can simply enable it as a plugin with
+the following configuration:
+
+```
+{
+    "plugins": ["./path-to-plugin.grit"]
+}
+```
+
+The plugin will now be enabled on all files the analyzer runs on.
+
+## Plugin API
+
+In addition to Grit's
+[built-in functions](https://docs.grit.io/language/functions#built-in-functions),
+Biome currently supports one extra function:
+
+### `register_diagnostic()`
+
+Registers a diagnostic to be reported whenever the pattern matches.
```

Comment:
The API doesn't mention the diagnostic category emitted by the plugin (error, warn, info). You should mention it and explain if users can control it and, if so, how. (or if they can't)

Reply:
Huh, it always uses the default severity... which I guess is error? That might be a nice thing to make configurable before 2.0 still :)

Reply:
Feature added: https://github.com/biomejs/biome/pull/5363
And documentation added on the page here.